internal/Picker/Picker.js

```diff
@@ -86,611 +87,244 @@ const allowedClassNames = ['picker', 'valueWrapper', 'joined', 'horizontal', 've
+	const handleDecrement = useCallback(() => {
+		if (!hasReachedBound(-step)) {
+			updateValue(-1);
+			setPressedState(-1);
 		}
-	};
+	}, [hasReachedBound, setPressedState, step, updateValue]);
 
-	handleDown = () => {
-		const {changedBy, joined, orientation} = this.props;
-
-		if (joined && orientation === 'horizontal' && changedBy === 'enter') {
-			this.setIncPickerButtonPressed();
+	const handleIncrement = useCallback(() => {
+		if (!hasReachedBound(step)) {
+			updateValue(1);
+			setPressedState(1);
 		}
+	}, [hasReachedBound, setPressedState, step, updateValue]);
 
-		if (joined && this.pickerButtonPressed === 1) {
-			this.handleIncrement();
+	const throttleWheelDec = useMemo(() => new Job(handleDecrement, 100), [handleDecrement]);
 
-			if (orientation === 'vertical' || changedBy === 'arrow') {
-				this.emulateMouseUp.start();
-			}
-		} else if (joined && this.pickerButtonPressed === -1) {
-			this.handleDecrement();
-			this.emulateMouseUp.start();
-		}
-	};
+	const throttleWheelInc = useMemo(() => new Job(handleIncrement, 100), [handleIncrement]);
 
-	handleWheel = (ev) => {
-		const {step} = this.props;
-		forwardWheel(ev, this.props);
+	const handleWheel = useCallback((ev) => {
+		forwardWheel(ev, props);
 
-		const isContainerSpotted = this.containerRef.current === Spotlight.getCurrent();
+		const isContainerSpotted = containerRef.current === Spotlight.getCurrent();
 
 		if (isContainerSpotted) {
 			const dir = -Math.sign(ev.deltaY);
 
 			// We'll sometimes get a 0/-0 wheel event we need to ignore or the wheel event has reached
 			// the bounds of the picker
-			if (dir && !this.hasReachedBound(step * dir)) {
+			if (dir && !hasReachedBound(step * dir)) {
 				// fire the onChange event
 				if (dir > 0) {
-					this.throttleWheelInc.throttle();
+					throttleWheelInc.throttle();
 				} else if (dir < 0) {
-					this.throttleWheelDec.throttle();
+					throttleWheelDec.throttle();
 				}
 				// simulate mouse down
-				this.setPressedState(dir);
+				setPressedState(dir);
 				// set a timer to simulate the mouse up
-				this.emulateMouseUp.start();
+				emulateMouseUp.start();
 				// prevent the default scroll behavior to avoid bounce back
 				ev.preventDefault();
 				ev.stopPropagation();
 			}
 		}
-	};
+	}, [emulateMouseUp, hasReachedBound, props, setPressedState, step, throttleWheelDec, throttleWheelInc]);
 
-	throttleWheelInc = new Job(this.handleIncrement, 100);
+	useEffect(() => {
+		const currentPicker = containerRef.current;
+		if (platform.webos) currentPicker.addEventListener('webOSVoice', handleVoice);
 
-	throttleWheelDec = new Job(this.handleDecrement, 100);
+		if (joined) {
+			currentPicker.addEventListener('wheel', handleWheel);
+		} else {
+			currentPicker.removeEventListener('wheel', handleWheel);
+		}
 
-	setDecPickerButtonPressed = () => {
-		this.pickerButtonPressed = -1;
-	};
+		return (() => {
+			emulateMouseUp.stop();
+			throttleWheelInc.stop();
+			throttleWheelDec.stop();
```

Comment:
About these stop functions, I think we need to put these in a separate cleanup of useEffect with dependency [].
WDYT?

Reply:
Yes, I agree.